src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,55 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        writeToFile(createReport(fromFileName), toFileName);
+    }
+
+    private String readFromFile(String fileName) {
+        try {
+            return Files.readString(Path.of(fileName));
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read from file " + fileName, e);
+        }
+    }
+
+    private String createReport(String dataFromFile) {
```

Comment:
I agree that createReport should accept dataFromFile, but you're actually passing filename, and calling read method inside this one. Let's separate our concerns and make methods more independent and flexible

Reply:
I hope I understood it correctly. done (I hope),